src/compiler/program.ts

```diff
@@ -3508,11 +3526,58 @@ export function createProgram(rootNamesOrOptions: readonly string[] | CreateProg
         return result;
     }
 
-    function getCreateSourceFileOptions(fileName: string, moduleResolutionCache: ModuleResolutionCache | undefined, host: CompilerHost, options: CompilerOptions): CreateSourceFileOptions {
+    function getImpliedNodeFormatForFile(fileName: string) {

```

Comment:
I’m open to suggestions for a different name to differentiate from `ts.getImpliedNodeFormatForFile` which is public API, but I intend to expose this one as `program.getImpliedNodeFormatForFile`, and add JSDoc to the top-level standalone one that says you should use the `program` version if possible.